tests/integration/agents/k8s/test_k8s_agent.py

```diff
@@ -0,0 +1,314 @@
+            ),
+            "k8s_overview_query_tool",
+        ),
+        # Test case for pod logs query
+        (
+            KubernetesAgentState(
+                agent_messages=[],
+                messages=[
+                    SystemMessage(
+                        content="The user query is related to: {'resource_api_version': 'v1', 'resource_namespace': 'default', 'resource_kind': 'Pod'}"
+                    ),
+                    HumanMessage(
+                        content="What deployments exist in the 'kube-system' namespace?"
+                    ),
+                ],
+                subtasks=[
+                    {
+                        "description": "What deployments exist in the 'kube-system' namespace?",
+                        "task_title": "Check existing deployment",
+                        "assigned_to": "KubernetesAgent",
+                    }
+                ],
+                my_task=SubTask(
+                    description="What deployments exist in the 'kube-system' namespace?",
+                    task_title="Check existing deployment",
+                    assigned_to="KubernetesAgent",
+                ),
+                k8s_client=Mock(spec_set=IK8sClient),  # noqa
+                is_last_step=False,
+                remaining_steps=AGENT_STEPS_NUMBER,
+            ),
+            "k8s_query_tool",
+        ),
+        # Test case for namespace overview
+        (
+            KubernetesAgentState(
+                agent_messages=[],
+                messages=[
+                    SystemMessage(
+                        content="The user query is related to: {'resource_namespace': 'production'}"
+                    ),
+                    HumanMessage(content="Is there any issue with my namespace"),
+                ],
+                subtasks=[
+                    {
+                        "description": "Is there any issue with my namespace",
+                        "task_title": "Namespace overview",
+                        "assigned_to": "KubernetesAgent",
+                    }
+                ],
+                my_task=SubTask(
+                    description="Is there any issue with my namespace",
+                    task_title="Namespace overview",
+                    assigned_to="KubernetesAgent",
+                ),
+                k8s_client=Mock(spec_set=IK8sClient),  # noqa
+                is_last_step=False,
+                remaining_steps=AGENT_STEPS_NUMBER,
+            ),
+            "k8s_overview_query_tool",
+        ),
+        # Test case for namespace overview
+        (
+            KubernetesAgentState(
+                agent_messages=[],
+                messages=[
+                    SystemMessage(
+                        content="The user query is related to: {'resource_namespace': 'production'}"
+                    ),
+                    HumanMessage(content="Show me cluster-level resource usage"),
```

Comment:
your query is about cluster, but the system message has the namespace. This is interesting case we need to test with the running companion

Reply:
have you tested this one?